overlord/devicestate/handlers_install.go

```diff
@@ -618,3 +634,72 @@ func (m *DeviceManager) doRestartSystemToRunMode(t *state.Task, _ *tomb.Tomb) er
 
 	return nil
 }
+
+var seedOpen = seed.Open
+
+var maybeApplyPreseededData = func(ubuntuSeedDir, sysLabel, writableDir string) (preseeded bool, err error) {
+	preseedArtifact := filepath.Join(ubuntuSeedDir, "systems", sysLabel, "preseed.tgz")
+	if !osutil.FileExists(preseedArtifact) {
+		return false, nil
+	}
+
+	// TODO: verify artifact signature?
```

Comment:
this overall looks good as first approximation, but we cannot really land it into master until we have solved this because otherwise is a big way to inject things into a system